crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -1085,7 +1085,13 @@ impl Analyzer<'_, '_> {
 
         match (to, rhs) {
             (_, Type::Conditional(rc)) => {
-                self.assign_with_opts(data, to, &rc.true_type, opts)
+                let ty = if rc.true_type.type_eq(&rc.check_type) {
```

Comment:
Is this logic correct? Not sure why is `Foo extends Bar ? Foo : Baz` special

Reply:
this pr is incomplete

reason is simple
```ts
type Bar = string;
function Test1<T extends Bar>(arg1 : T){	
	const some : Bar = arg1; // ok
  arg1 = "123" as Bar; // error
}
```

Reply:
additional..

if check_type same true_type, true_type has extends_type
because check_type extends extends_type

1. check_type == true_type
2. check_type = extends_type
3. true_type = extends_type

Reply:
Of course, I'd like to visit the lower type whistle,
but I forgot the name of the function name.